core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SharedHttpClientSessionManager.java

```diff
@@ -66,7 +63,8 @@
 
 	public SharedHttpClientSessionManager() {
 		final ThreadFactory backingThreadFactory = Executors.defaultThreadFactory();
-		this.executor = Executors.newScheduledThreadPool(cores, (Runnable runnable) -> {
+		final Integer corePoolSize = Integer.getInteger(CORE_POOL_SIZE_PROPERTY, 1);
```

Comment:
Shouldn't this use `System.getProperty` to get the _value_ of the property?

Reply:
It does, behind the scenes. Just have a look at the `Integer#getInteger` method! Using this approach avoids the boilerplate code required to cast a `String` to an `Integer`. But I see now that the `corePoolSize` variable should be `int`, and not the nullable `Integer`. I'll fix.